File: C++/2966. Divide Array Into Arrays With Max Difference.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    bool mutualDiff(int a, int b, int c, int k)
    {
        if (a - b > k || b - c > k || a - c > k)
            return true;
        return false;
    }
    vector<vector<int>> divideArray(vector<int>& nums, int k)
    {
        vector<vector<int>>ans;
        sort(nums.begin(), nums.end());
        for (int i = 0;i < nums.size();i += 3)
            if (mutualDiff(nums[i + 2], nums[i + 1], nums[i], k))
                return ans;

        for (int i = 0;i < nums.size();i += 3)
            ans.push_back({ nums[i],nums[i + 1],nums[i + 2] });
        return ans;
    }
};

File: Python/2966. Divide Array Into Arrays With Max Difference.py
class Solution:
    def mutualDiff(self, a, b, c, k):
        if a - b > k or b - c > k or a - c > k:
            return True
        return False

    def divideArray(self, nums, k):
        ans = []
        nums.sort()

        for i in range(0, len(nums), 3):
            if self.mutualDiff(nums[i + 2], nums[i + 1], nums[i], k):
                return ans

        for i in range(0, len(nums), 3):
            ans.append([nums[i], nums[i + 1], nums[i + 2]])

        return ans